src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java

```diff
@@ -184,9 +185,19 @@ public boolean accept(TreeWalkerAuditEvent event) {
      * @return true if it is matching
      */
     private boolean isFileNameMatching(TreeWalkerAuditEvent event) {
-        return event.getFileName() != null
-                && ((event.getFileName()).equals(fileName)
-                || event.getFileName().contains(fileName));
+        String eventFileName = event.getFileName();
+        boolean result = eventFileName != null;
+
+        if (result) {
+            // git always displays paths with '/', even on windows
+            if (File.separatorChar != '/') {
+                eventFileName = eventFileName.replace(File.separatorChar, '/');
+            }
+
+            result = eventFileName.endsWith(fileName);
```

Comment:
I forgot to mention I changed `equals`/`contains` to `endsWith`.
I could understand `equals` which I can restore if needed, but `contains` seems like it could lead to false positives more than `endsWith` can.
There should be some better method than this, but without knowing the base directory location, we are stuck with this kind of matching.

Reply:
Is it possible to cover this code changes with test ?
We already have top level tests that takes patch, file, config and run validation.

So patch will be in windows style, with windows line endings. We will have line it got settings that such file should be windows. We can set some system properties  during test execution mimic windows.